anndata/_core/merge.py

```diff
@@ -245,6 +245,29 @@ def try_unifying_dtype(col: list) -> pd.core.dtypes.base.ExtensionDtype | None:
         return None
 
 
+def check_combinable_cols(cols: list[pd.Index], join: Literal["inner", "outer"]):
+    """Given columns for a set of dataframes, checks if the can be combined.
+
+    Looks for if there are duplicated column names that would show up in the result.
+    """
+    if join == "inner":
+        repeated_cols = reduce(lambda x, y: x.union(y[y.duplicated()]), cols, set())
+        intersecting_cols = intersect_keys(cols)
+        problem_cols = repeated_cols.intersection(intersecting_cols)
+    elif join == "outer":
+        problem_cols = reduce(lambda x, y: x.union(y[y.duplicated()]), cols, set())
+    else:
+        raise ValueError()
```

Comment:
Please don’t duplicate complex chunks of code. I played “find the difference” for a few seconds because two identical variable assignments should mean “there’s a subtle difference between those lines”

```suggestion
    repeated_cols = reduce(lambda x, y: x.union(y[y.duplicated()]), cols, set())
    if join == "inner":
        intersecting_cols = intersect_keys(cols)
        problem_cols = repeated_cols.intersection(intersecting_cols)
    elif join == "outer":
        problem_cols = repeated_cols
    else:
        raise ValueError()
```